src/dataflow/core/linearize.py

```diff
@@ -12,12 +12,24 @@
     Lispress,
     lispress_to_program,
     program_to_lispress,
+    render_compact,
 )
 from dataflow.core.program import Program
 from dataflow.core.program_utils import Idx, OpType
 from dataflow.core.sexp import Sexp
 
 
+def to_canonical_form(tokenized_lispress: str) -> str:
+    """Returns canonical form of a tokenized lispress.
+
+    The canonical form is un-tokenized and compact; it also sorts named arguments in alphabetical order.
+    """
+    lispress = seq_to_lispress(tokenized_lispress.split(" "))
+    program, _ = lispress_to_program(lispress, 0)
+    round_tripped = program_to_lispress(program)
+    return render_compact(round_tripped)
```

Comment:
is this the same as?
```suggestion
    deformatted = sexp_deformatter(tokenized_lispress.split(" "))
    return try_round_trip(deformatted)
```

Reply:
Hmm... the type doesn't match?
`sexp_deformatter` returns `List[str]`, but `try_round_trip` takes `str`.
We may change change `try_round_trip` to take `Lispress` by moving `parse_lispress(lispress_str)` outside the `try_round_trip`, then we can reduce some lines of the code here.

Reply:
oh yeah, sorry, wrote without testing. no, i want the parse_lispress inside the try. i think it's fine to leave it how you have it